src/silx/gui/plot/items/_roi_base.py

```diff
@@ -437,6 +441,26 @@ def setVisible(self, visible):
             self._visible = visible
             self._updated(items.ItemChangedType.VISIBLE)
 
+    def getText(self) -> str:
+        """Returns the currently displayed text for this ROI"""
+        return self.getName() if self.__text is None else self.__text
```

Comment:
I would prefer to decouple the `text` property from the `name`.

You could see the use of the name as a display mode (which we don't have to handle explicitly).

What do you think?

But i dont see any problem with the actual implementation.

Reply:
What do you mean? Having `getText` eventually returning `self.__text` and an extra method, e.g., `getDisplayedText` to retrieve what should be displayed?

Reply:
After discussion, this is fine as it is